File: samples/sample_plugin.test.js
// Test ultra avancé clé en main pour SamplePlugin JS
const SampleThreedPlugin = require('./sample_plugin');

describe('SampleThreedPlugin (JS)', () => {
  let plugin;
  beforeEach(() => {
    plugin = new SampleThreedPlugin();
  });
  test('run retourne un objet enrichi', () => {
    const result = plugin.run({ foo: 'bar' });
    expect(result).toHaveProperty('plugin', 'SampleThreedPlugin');
    expect(result).toHaveProperty('status', 'processed');
  });
  // ... edge cases, hooks, audit, sécurité, CI/CD
});

File: samples/sample_plugin.test.py
# Test ultra avancé clé en main pour SamplePlugin Python
import unittest
from backend.components.metiers.threed.plugins.core.sample_plugin import SamplePlugin

class TestSamplePlugin(unittest.TestCase):
    def setUp(self):
        self.plugin = SamplePlugin()
    def test_run(self):
        result = self.plugin.run({'foo': 'bar'})
        self.assertIn('Traitement avancé', result)
    def test_audit_trail(self):
        self.plugin.activate({'user': {'role': 'admin'}})
        self.plugin.run({'foo': 'bar'})
        self.assertGreater(len(self.plugin.get_audit_trail()), 0)
    def test_activation_non_admin(self):
        with self.assertRaises(Exception):
            self.plugin.activate({'user': {'role': 'guest'}})
    # ... edge cases, hooks, audit, sécurité, CI/CD

if __name__ == '__main__':
    unittest.main()
